static-assets/components/cstudio-forms/controls/rte.js

```diff
@@ -279,248 +283,263 @@ CStudioAuthoring.Module.requireModule(
+                }
+              },
+              ...(rteConfig?.tinymceOptions && {
+                ...CrafterCMSNext.util.object.reversePluckProps(
+                  // Tiny seems to somehow mutate the options object which would cause crashes when attempting
+                  // to mutate immutable object (possibly from redux). Also, we don't want the state to get mutated.
+                  JSON.parse(JSON.stringify(rteConfig.tinymceOptions)),
+                  'target', // Target can't be changed
+                  'inline', // The control will always have the default (false) in forms-engine.
+                  'setup',
+                  'base_url',
+                  'encoding',
+                  'autosave_ask_before_unload', // Autosave options are removed since it is not supported in control.
+                  'autosave_interval',
+                  'autosave_prefix',
+                  'autosave_restore_when_empty',
+                  'autosave_retention',
+                  'file_picker_callback', // File picker integration with our data sources is already implemented in the control
+                  'height', // Height is set via control properties
+                  'width', // Width is set via control properties
+                  'paste_postprocess', // Already implemented for paste and drag&drop using our data sources.
+                  'images_upload_handler', // Images upload integration with our data sources is already implemented in the control
+                  'code_editor_inline', // Code editor will always be inline in forms-engine.
+                  'plugins', // Considered/used above, mixed with our options
+                  'external_plugins', // Considered/used above, mixed with our options
+                  'relative_urls', // To avoid allowing convertion of urls to be relative to the document_base_url
+                  'readonly', // Comes from form control props, can't be overridden.
+                  'content_css' // Handled above, if no content_css is found it will use dark/default styles.
+                )
+              })
+            });
+
+            const isInRepeatGroup = Boolean(config.repeatContainer);
+            // Update all content before saving the form (all content is automatically updated on focusOut)
+            callback = {
+              beforeSave: function () {
+                _thisControl.save();
+              },
+              ...(isInRepeatGroup ? { repeatGroupId: config.repeatContainer.id } : {})
+            };
+            _thisControl.form.registerBeforeSaveCallback(callback);
           };
-          _thisControl.form.registerBeforeSaveCallback(callback);
+
+          if (config.onDemandEditorInitialization) {
+            onDemandEditorInitializationEl.addEventListener('click', () => {
+              initTiny();
+              onDemandEditorInitializationEl.style.display = 'none';
```

Comment:
Is there a reason not to remove it?
Could we maybe use the same element and avoid this function returning two elements?